en/identity-server/next/docs/references/extend/oauth2/write-a-custom-oauth-2.0-grant-type.md

```diff
@@ -0,0 +1,341 @@
+
+1. On the WSO2 Identity Server Console, go to **Applications**.
+
+2. Click **New Application** and select **Standard-Based Application**.
+
+3. Provide an application name and select **OIDC** as the protocol.
+
+4. Click **Register** to complete the registration.
+
+5. In the **Protocol** tab, select the custom grant type (mobile) from the **Allowed Grant Types** list.
+
+6. Click **Update**. Note the **Client ID** and **Client secret** that appear. 
+
+    !!! tip
+        To configure more advanced configurations, see [OIDC Configurations]({{base_path}}/references/app-settings/oidc-settings-for-app).
+
+### Obtain an access token
+
+Send the grant request to the `          /token         ` API using a cURL command.
+
+!!! note
+    The HTTP POST body must contain the following two parameters: `grant_type=mobile` and `mobileNumber`. 
+
+    ``` 
+    grant_type=mobile&mobileNumber=<MOBILE_NUMBER>
+    ```
+
+!!! abstract ""
+    **Request format**
+    ```
+    curl --user <OAUTH_CLIENT_KEY>:<OAUTH_CLIENT_SECRET> -k -d "grant_type=<CUSTOM_GRANT_TYPE>&mobileNumber=<MOBILE_NUMBER>" -H "Content-Type: application/x-www-form-urlencoded" https://<IS_HOST>:<IS_PORT>/oauth2/token
+    ```
+    ---
+    **Sample request**
+    ```curl
+    curl --user 7wYeybBGCVfLxPmS0z66WNMffyMa:WYfwHUsbsEvwtqmDLuaxF_VCQJwa -k -d "grant_type=mobile&mobileNumber=0333444" -H "Content-Type: application/x-www-form-urlencoded" https://localhost:9443/oauth2/token
+    ```
+
+You will receive a response similar to the following JSON response with the access token.
+
+```
+{
+    "token_type":"bearer",
+    "expires_in":2823,
+    "refresh_token":"26e1ebf16cfa4e67c3bf39d72d5c276",
+    "access_token":"d9ef87802a22cf7682c2e77df72c735"
+}
+```
+
+!!! note "Note"
+    The access token and the refresh token that are received with the response are just randomly generated by the sample application. Those tokens are not valid tokens.
+
+------------------------------------------------------------------------
+
+## Customize an existing grant type
+
+As an alternative to creating a custom OAuth grant type, you can
+customize one of the existing grant types. The following two classes are
+sample implementations of customizing the password grant type in
+particular but any other grant type can be customized as well.
+
+-   [RoleBasedPasswordGrant](https://github.com/wso2/samples-is/blob/v4.5.6/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/password/RoleBasedPasswordGrant.java) -
+    This does some RBAC validation apart from the authentication before
+    granting access.
+-   [ModifiedAccessTokenPasswordGrant](https://github.com/wso2/samples-is/blob/v4.5.6/oauth2/custom-grant/src/main/java/org/wso2/sample/identity/oauth2/grant/password/ModifiedAccessTokenPasswordGrant.java) -
+    This implementation customized the access token value.
+
+### Configure the resource
+
+1.  Copy the .jar file into the
+    `          <IS_HOME>/repository/component/lib         `
```

Comment:
should this be dropins?

Reply:
Same as https://github.com/wso2/docs-is/pull/4268#discussion_r1433443588